internal/middleware.go

```diff
@@ -0,0 +1,6 @@
+package internal
+
+import "net/http"
+
+// Middleware is a convenience type for functions that wrap http.Handlers.
+type Middleware func(http.Handler) http.Handler
```

Comment:
Rather than moving this around, wh not copy the code into the package you need it? 
This can easily stay in the `internal/authentication` package unmodified. 
Simply create a new one in `internal/ratelimit/gubernator`.

Reply:
Our middleware definition is the same for all so I want to share it. What is the value copying the same definition for each package?